File: backend/app/schemas/weight.py
from datetime import date, datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, Field


class WeightEntryCreate(BaseModel):
    weight: float = Field(..., gt=0, le=500)
    recorded_date: date
    note: str | None = Field(None, max_length=500)


class WeightEntryUpdate(BaseModel):
    weight: float | None = Field(None, gt=0, le=500)
    note: str | None = Field(None, max_length=500)


class WeightEntryResponse(BaseModel):
    id: UUID
    user_id: UUID
    weight: float
    recorded_date: date
    note: str | None = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class WeightHistoryResponse(BaseModel):
    entries: list[WeightEntryResponse]
    total: int
    start_weight: float | None = None
    current_weight: float | None = None
    weight_change: float | None = None

File: frontend/src/types/weight.ts
export interface WeightEntryCreate {
  weight: number;
  recorded_date: string;
  note?: string | null;
}

export interface WeightEntry {
  id: string;
  user_id: string;
  weight: number;
  recorded_date: string;
  note: string | null;
  created_at: string;
  updated_at: string;
}

export interface WeightHistoryResponse {
  entries: WeightEntry[];
  total: number;
  start_weight: number | null;
  current_weight: number | null;
  weight_change: number | null;
}
